docs/site/src/plugins/inject-code/injectLoader.js

```diff
@@ -0,0 +1,204 @@
+          );
+          let fileExt = injectFile.substring(injectFile.lastIndexOf(".") + 1);
+          let language = "";
+          const fullPath = path.join(repoPath, injectFile);
+
+          // Assuming rust manifests and locks never being used
+          switch (fileExt) {
+            case "toml":
+              language = "move";
+              break;
+            case "lock":
+              language = "move";
+              break;
+            case "sh":
+              language = "shell";
+              break;
+            case "mdx":
+              language = "markdown";
+              break;
+            case "tsx":
+              language = "ts";
+              break;
+            default:
+              language = fileExt;
+          }
+
+          if (fs.existsSync(fullPath)) {
+            let injectFileContent = fs.readFileSync(fullPath, "utf8");
+            const marker =
+              injectFileFull.indexOf("#") > 0
+                ? injectFileFull.substring(injectFileFull.indexOf("#"))
+                : null;
+            if (marker) {
+              const regexStr = `\\/\\/\\s?docs::${marker.trim()}\\b([\\s\\S]*)\\/\\/\\s*docs::\\/\\s?${marker.trim()}\\b`;
+              const closingsStr = `\\/\\/\\s?docs::\\/${marker.trim()}\\b([)};]*)`;
+
+              const closingRE = new RegExp(closingsStr, "g");
+
+              const regex = new RegExp(regexStr, "g");
+              const match = regex.exec(injectFileContent);
+              const closingStr = closingRE.exec(injectFileContent);
+              if (match) {
+                injectFileContent = match[1];
+              }
+              if (closingStr) {
+                const closingTotal = closingStr[1].length;
+                let closingArray = [];
+                for (let i = 0; i < closingTotal; i++) {
+                  const currentChar = closingStr[1][i];
+                  const nextChar = closingStr[1][i + 1];
+
+                  if (nextChar === ";") {
+                    closingArray.push(currentChar + nextChar);
+                    i++;
+                  } else {
+                    closingArray.push(currentChar);
+                  }
+                }
+                const totClosings = closingArray.length;
+
+                // Process any closing elements added in the closing comment of source code
+                let closing = "";
+                for (let j = 0; j < totClosings; j++) {
+                  let space = "  ".repeat(totClosings - 1 - j);
+                  closing += `\n${space}${closingArray[j]}`;
+                }
+                injectFileContent = injectFileContent.trim() + closing;
+              }
+
+              // Start searching for the pause doc comments
+              // Must be in form: // docs::#idName-pause: optional replacement text
+              // Must be followed at some point by: // docs::#idName-resume
+              // Replace text in the middle with optional replacement text or blankness
```

Comment:
Rather than having special tags for pausing and resuming the injection of a specific example, it may be neater (in terms of in the code being annotated) to have annotations for anchor points in the code, and then require that the inject invocation lists out multiple anchor points if it wants to have an example that consists of multiple disjoint snippets?

The syntax for injection would also need to deal with the placeholder syntax and closing braces in that case (rather than the syntax for annotations), but this makes sense to me, because if you are using the same snippet of code in multiple injections, you may want to use a different placeholder in each one (e.g. in one case, you use `// TODO`, in another case you use `// ... snip ...` etc).